### ISSUE TITLE
Fix reward epoch processor to respect -1 end block

### ISSUE THREAD

**Issue by pharr117** [DefiantLabs/cosmos-tax-cli#380]
The reward processor does not respect the -1 value, instead setting the end block to the latest block height when it spins up.

We will need to rework the logic so that it can handle long periods of sleeping to wait for the next reward epoch to be written so that it can allow the app to stay alive in the background long term.

The code in question is found here:

https://github.com/DefiantLabs/cosmos-tax-cli/blob/c3e003ae4b1ea1fa186d0168ba1b27559eace1a5/cmd/index.go#L386-L394

**Comment by danbryan** [comment]
does it have to be a separate indexer? Can't it just be like.

check block 1
found transactions
didn't find rewards
index 1 transactions

check block  2
found transactions
found rewards
index 2 transactions
index 2 rewards

check block 3
......

**Comment by pharr117** [comment]
TL;DR no it doesn't HAVE to be separate indexers, but it would be a bit complex to rework it now and may slow the app down considerably.

That is a good question to ask, but its a complicated answer for a number of reasons:

1. We have separate Golang parallel processing of transaction indexing and reward processing. This allows the application to run faster
2. Moving reward processing into the same threads for transaction processing will slow the app down considerably since it will need to work on the same block at the same time.
    * According to a code comment here, querying blocks for this info is super slow:
    https://github.com/DefiantLabs/cosmos-tax-cli/blob/c3e003ae4b1ea1fa186d0168ba1b27559eace1a5/osmosis/rewards.go#L86-L89
    * This is probably why Kyle made the decision to split them out
3. We would need to rework how the configuration options work. The separate indexers work off separate configuration options, particularly the start and end blocks. We would want to unify these.

At a quick glance, right now the rewards indexing is all Osmosis rewards oriented, its not very general to index Epochs events on other chains.

**Comment by danbryan** [comment]
@pharr117  lets do a voice chat about this during the sprint today
